src/query/execute_schema.c

```diff
@@ -4573,12 +4577,24 @@ do_rename_partition (MOP old_class, const char *newname)
 
   newlen = strlen (newname);
 
-  error = au_fetch_class (old_class, &smclass, AU_FETCH_READ, AU_SELECT);
+  error = au_fetch_class (old_class, &smclass, AU_FETCH_UPDATE, AU_ALTER);
   if (error != NO_ERROR)
     {
       goto end_rename;
     }
 
+  strncpy (expr_copy, smclass->partition->expr, DB_MAX_PARTITION_EXPR_LENGTH);
+  keycol = expr_copy + 7;	/* strlen ("SELECT") + empty_string = 7 */
+  keycol_end = strstr (keycol, " ");
+  keycol_end[0] = '\0';
+
+  assert (snprintf (NULL, 0, "SELECT %s FROM [%s]", keycol, newname) < DB_MAX_PARTITION_EXPR_LENGTH);
```

Comment:
In release mode, unexpected behavior could occur, How about `return ER_FAILED` instead of assert?

Reply:
Removed assert() check from commit in "Change only after FROM in pexpr. (2bcc3b8)".

Since DB_MAX_PARTITION_EXPR_LENGTH has a length of 2048 and newname has a maximum length of 255,
I thought the length didn't need to be checked.